spec/models/user_spec.rb

```diff
@@ -1,5 +1,18 @@
 require 'rails_helper'
 
 RSpec.describe User, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
-end
+  context "when a user is created it" do
+    it "can be created with just an email parameter" do
+      u1 = User.create(email: "name@blah.com")
+      expect(u1).to be_an_instance_of(User)
+      expect(u1.email).to eq("name@blah.com")
+    end
+    it "can generate a password" do
+      u2 = User.create(email: "name2@blah.com")
+      u2.generate_password
```

Comment:
Calling `.create` already invokes `generate_password`.  To test `generate_password` explicitly like this, use `User.new` which does not make that call for you.

If you really want to be pedantic about it, you could `expect(u2.password).to be_nil` prior to the `u2.generate_password` line.